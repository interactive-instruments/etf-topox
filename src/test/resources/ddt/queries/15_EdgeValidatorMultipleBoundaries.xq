declare namespace adv='http://www.adv-online.de/namespaces/adv/gid/6.0';
declare namespace wfsAdv='http://www.adv-online.de/namespaces/adv/gid/wfs';
declare namespace gml='http://www.opengis.net/gml/3.2';
declare namespace ete='http://www.interactive-instruments.de/etf/topology-error/1.0';
declare namespace uuid='java.util.UUID';

import module namespace topox = 'https://modules.etf-validator.net/topox/1';


declare function local:log($text as xs:string) as empty-sequence()
{
  prof:dump($text)
};

declare variable $dbname external := 'TOPOX-JUNIT-TEST-DB-000';
declare variable $tmpOutputDirectory external := 'tmp/bsx';

declare variable $modellart external := "DLKM";

let $db := db:open(topox:init-db($dbname, xs:short(1)))
let $dbCount := 3
(: Calculated factor. Must be verified in tests :)
let $initialEdgeCapacity := xs:int($dbCount * 1995000)


let $surfaceLocalNames := ('AX_Fliessgewaesser', 'AX_Hafenbecken', 'AX_Meer', 'AX_StehendesGewaesser', 'AX_Bergbaubetrieb', 'AX_FlaecheBesondererFunktionalerPraegung', 'AX_FlaecheGemischterNutzung', 'AX_Friedhof', 'AX_Halde', 'AX_IndustrieUndGewerbeflaeche', 'AX_Siedlungsflaeche', 'AX_SportFreizeitUndErholungsflaeche', 'AX_TagebauGrubeSteinbruch', 'AX_Wohnbauflaeche', 'AX_FlaecheZurZeitUnbestimmbar', 'AX_Gehoelz', 'AX_Heide', 'AX_Landwirtschaft', 'AX_Moor', 'AX_Sumpf', 'AX_UnlandVegetationsloseFlaeche', 'AX_Wald', 'AX_Bahnverkehr', 'AX_Flugverkehr', 'AX_Platz', 'AX_Schiffsverkehr', 'AX_Strassenverkehr', 'AX_Weg')
let $surfaces := $db/*:AX_Bestandsdatenauszug/*:enthaelt/wfsAdv:FeatureCollection/gml:featureMember/*[local-name() = $surfaceLocalNames]

let $curves := $db/*:AX_Bestandsdatenauszug/*:enthaelt/wfsAdv:FeatureCollection/gml:featureMember/*[local-name() = "AX_Gebietsgrenze"][*:artDerGebietsgrenze = ('7101','7102')]

let $logMessage1 := local:log("Initializsing TopoX " || topox:detailed-version() || ". This may take a while...")
let $initTime := prof:current-ms()

let $topoId := topox:new-topology(
  'Tatsaechliche_Nutzung',
  $tmpOutputDirectory,
  $initialEdgeCapacity
)

let $duration := prof:current-ms()-$initTime
let $dummy := local:log("TopoX initialized in " || $duration || " ms" )

let $initTime := prof:current-ms()
let $dummy := topox:parse-surface($surfaces, 'adv:position/gml:Surface', $topoId)
let $duration := prof:current-ms()-$initTime
let $dummy := local:log("Topology built in " || $duration || " ms" )

let $boundaryId := topox:new-validator-multiple-boundaries($topoId)
let $borderParsingTime := prof:current-ms()
(: Der zweite Parameter wird derzeit ignoriert, ggfs. in Zukunft Funktion nutzen. :)
let $borderParsingDummy := topox:parse-boundary($curves, '*:position/gml:*', $boundaryId)
let $borderParsingDuration := prof:current-ms()-$borderParsingTime
let $borderParsingLog := local:log("Borders built in " || $borderParsingDuration || " ms" )

let $errorCodes := ('POINT_DETACHED', 'EDGE_POINTS_INVALID')
let $errors := topox:topological-errors($topoId, $errorCodes)


(:
  To view file on local machine, temporary toggle
  security.fileuri.strict_origin_policy to false in Firefox
:)
let $initTime := prof:current-ms()
let $dummy := topox:export-erroneous-features-to-geojson($topoId, "Map_" || uuid:randomUUID() || ".js")
let $duration := prof:current-ms()-$initTime
let $dummy := local:log(" Results exported in " || $duration || " ms" )

return topox:topological-errors-doc($topoId)

