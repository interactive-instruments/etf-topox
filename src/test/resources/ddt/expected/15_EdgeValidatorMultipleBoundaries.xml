<?xml version="1.0" encoding="UTF-8"?><ete:TopologicalErrors xmlns:ete="http://www.interactive-instruments.de/etf/topology-error/1.0" name="Tatsaechliche_Nutzung"><e i="1" t="POINT_DETACHED"><X>678316.44</X><Y>5665046.92</Y><IS>124554053030</IS></e><e i="2" t="POINT_DETACHED"><X>678315.8</X><Y>5665039.53</Y><IS>124554053030</IS></e><e i="3" t="POINT_DETACHED"><X>678435.09</X><Y>5665007.02</Y><IS>137438954921</IS></e><e i="4" t="POINT_DETACHED"><X>678400.3</X><Y>5664905.56</Y><IS>150323856812</IS></e><e i="5" t="POINT_DETACHED"><X>678308.06</X><Y>5664924.32</Y><IS>163208758703</IS></e><e i="6" t="POINT_DETACHED"><X>678305.82</X><Y>5664891.51</Y><IS>176093660594</IS></e><e i="7" t="POINT_DETACHED"><X>678302.64</X><Y>5664869.34</Y><IS>188978562485</IS></e><e i="8" t="POINT_DETACHED"><X>678297.72</X><Y>5664844.87</Y><IS>201863464376</IS></e><e i="9" t="POINT_DETACHED"><X>678292.22</X><Y>5664824.96</Y><IS>214748366267</IS></e><e i="10" t="POINT_DETACHED"><X>678289.08</X><Y>5664815.76</Y><IS>227633268158</IS></e><e i="11" t="POINT_DETACHED"><X>678284.98</X><Y>5664803.24</Y><IS>240518170049</IS></e><e i="12" t="POINT_DETACHED"><X>678282.38</X><Y>5664791.0</Y><IS>253403071940</IS></e><e i="13" t="POINT_DETACHED"><X>678281.96</X><Y>5664771.18</Y><IS>266287973831</IS></e><e i="14" t="POINT_DETACHED"><X>678279.16</X><Y>5664756.39</Y><IS>279172875722</IS></e><e i="15" t="POINT_DETACHED"><X>678274.68</X><Y>5664747.58</Y><IS>292057777613</IS></e><e i="16" t="POINT_DETACHED"><X>678266.77</X><Y>5664735.36</Y><IS>304942679504</IS></e><e i="17" t="POINT_DETACHED"><X>678261.06</X><Y>5664728.32</Y><IS>317827581395</IS></e><e i="18" t="POINT_DETACHED"><X>678261.36</X><Y>5664724.74</Y><IS>330712483286</IS></e><e i="19" t="POINT_DETACHED"><X>678263.2</X><Y>5664721.52</Y><IS>343597385177</IS></e><e i="20" t="POINT_DETACHED"><X>678267.94</X><Y>5664719.03</Y><IS>356482287068</IS></e><e i="21" t="POINT_DETACHED"><X>678283.45</X><Y>5664713.08</Y><IS>369367188959</IS></e><e i="22" t="POINT_DETACHED"><X>678288.15</X><Y>5664711.42</Y><IS>382252090850</IS></e><e i="23" t="POINT_DETACHED"><X>678290.92</X><Y>5664709.47</Y><IS>395136992741</IS></e><e i="24" t="POINT_DETACHED"><X>679212.23</X><Y>5665415.17</Y><IS>124554053402</IS></e><e i="25" t="POINT_DETACHED"><X>679073.15</X><Y>5665038.57</Y><IS>124554053402</IS></e><e i="26" t="POINT_DETACHED"><X>678802.95</X><Y>5665093.34</Y><IS>137438955293</IS></e><e i="27" t="POINT_DETACHED"><X>678861.47</X><Y>5665270.35</Y><IS>150323857184</IS></e><e i="28" t="POINT_DETACHED"><X>678882.13</X><Y>5665328.45</Y><IS>163208759075</IS></e><e i="29" t="POINT_DETACHED"><X>678813.75</X><Y>5665346.55</Y><IS>176093660966</IS></e><e i="30" t="POINT_DETACHED"><X>678802.37</X><Y>5665347.71</Y><IS>188978562857</IS></e><e i="31" t="POINT_DETACHED"><X>678788.82</X><Y>5665351.53</Y><IS>201863464748</IS></e><e i="32" t="POINT_DETACHED"><X>678737.86</X><Y>5665364.55</Y><IS>214748366639</IS></e><e i="33" t="POINT_DETACHED"><X>678680.77</X><Y>5665372.61</Y><IS>227633268530</IS></e><e i="34" t="POINT_DETACHED"><X>678609.84</X><Y>5665383.03</Y><IS>240518170421</IS></e><e i="35" t="POINT_DETACHED"><X>678534.06</X><Y>5665388.63</Y><IS>253403072312</IS></e><e i="36" t="POINT_DETACHED"><X>678493.39</X><Y>5665388.61</Y><IS>266287974203</IS></e><e i="37" t="POINT_DETACHED"><X>678495.69</X><Y>5665371.82</Y><IS>279172876094</IS></e><e i="38" t="POINT_DETACHED"><X>678477.65</X><Y>5665299.38</Y><IS>292057777985</IS></e><e i="39" t="POINT_DETACHED"><X>678386.12</X><Y>5665321.97</Y><IS>304942679876</IS></e><e i="40" t="POINT_DETACHED"><X>678329.12</X><Y>5665336.04</Y><IS>317827581767</IS></e><e i="41" t="POINT_DETACHED"><X>678327.65</X><Y>5665301.69</Y><IS>330712483658</IS></e><e i="42" t="POINT_DETACHED"><X>678326.66</X><Y>5665238.23</Y><IS>343597385549</IS></e><e i="43" t="POINT_DETACHED"><X>678318.78</X><Y>5665071.85</Y><IS>356482287440</IS></e><e i="44" t="POINT_DETACHED"><X>678317.71</X><Y>5665058.98</Y><IS>369367189331</IS></e><e i="45" t="POINT_DETACHED" p="1"><X>678316.44</X><Y>5665046.92</Y><IS>382252091222</IS></e><e i="46" t="EDGE_POINTS_INVALID"><X>673971.7</X><Y>5664257.73</Y><IS>201863465027</IS><X2>673988.4</X2><Y2>5664161.65</Y2></e></ete:TopologicalErrors>